administrator/components/com_menus/src/Model/ItemModel.php

```diff
@@ -1426,7 +1426,7 @@ public function rebuild()
 	 */
 	public function save($data)
 	{
-		$pk         = (!empty($data['id'])) ? $data['id'] : (int) $this->getState('item.id');
+		$pk = (!empty($data['id']) || $data['id'] === 0) ? $data['id'] : (int) $this->getState('item.id');
```

Comment:
```suggestion
		$pk         = (!empty($data['id']) || $data['id'] === 0) ? $data['id'] : (int) $this->getState('item.id');
```
Keep alignment of "=" to the one in the line below like it was before.

Reply:
Another problem which I see with the changed code is that before the change, the `empty($data['id'])` did also catch the case when the array element $data['id'] is not set. Now when this is the case, we will run into the 2nd part of the condition, the `$data['id'] === 0` check, and this for sure will raise a PHP warning when the array element is not set.

See the PHP documentation of `empty` here: https://www.php.net/manual/en/function.empty.php

> No warning is generated if the variable does not exist. That means empty() is essentially the concise equivalent to !isset($var) || $var == false.

Reply:
sure! Add commit soon. I didn't notice it. thanks :)

Reply:
> Another problem which I see with the changed code is that before the change, the `empty($data['id'])` did also catch the case when the array element $data['id'] is not set. Now when this is the case, we will run into the 2nd part of the condition, the `$data['id'] === 0` check, and this for sure will raise a PHP warning when the array element is not set.
> 
> See the PHP documentation of `empty` here: https://www.php.net/manual/en/function.empty.php
> 
> > No warning is generated if the variable does not exist. That means empty() is essentially the concise equivalent to !isset($var) || $var == false.

ok sure! I'll look into it.

Reply:
Regarding how to fix the thing with the `$data['id']` I could tell you how to do it, but maybe you want to find it out yourself, so I won't advise unless you ask me for it.

Reply:
> Regarding how to fix the thing with the `$data['id']` I could tell you how to do it, but maybe you want to find it out yourself, so I won't advise unless you ask me for it.

If u know then please give some advise. So i can solve it soon. I want to solve it my own but i m busy in completing task assign regarding project 1 Joomla.

Reply:
Well I first have to think about if your fix is really right for all scenarios which we have.

Reply:
Sure!

Reply:
I think you should do it like it is already done in the admin model: Just use `isset($data['id'])` instead of `(!empty($data['id']) || $data['id'] === 0)` like it is done here https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/MVC/Model/AdminModel.php#L1353 .